NorthstarDLL/squirrel/squirrel.cpp

```diff
@@ -157,57 +157,6 @@ const char* SQTypeNameFromID(int type)
 	return "";
 }
 
-template <ScriptContext context> void SquirrelManager<context>::GenerateSquirrelFunctionsStruct(SquirrelFunctions* s)
-{
-	s->RegisterSquirrelFunc = RegisterSquirrelFunc;
-	s->__sq_defconst = __sq_defconst;
-
-	s->__sq_compilebuffer = __sq_compilebuffer;
-	s->__sq_call = __sq_call;
-	s->__sq_raiseerror = __sq_raiseerror;
-	s->__sq_compilefile = __sq_compilefile;
-
-	s->__sq_newarray = __sq_newarray;
-	s->__sq_arrayappend = __sq_arrayappend;
-
-	s->__sq_newtable = __sq_newtable;
-	s->__sq_newslot = __sq_newslot;
-
-	s->__sq_pushroottable = __sq_pushroottable;
-	s->__sq_pushstring = __sq_pushstring;
-	s->__sq_pushinteger = __sq_pushinteger;
-	s->__sq_pushfloat = __sq_pushfloat;
-	s->__sq_pushbool = __sq_pushbool;
-	s->__sq_pushasset = __sq_pushasset;
-	s->__sq_pushvector = __sq_pushvector;
-	s->__sq_pushobject = __sq_pushobject;
-
-	s->__sq_getstring = __sq_getstring;
-	s->__sq_getinteger = __sq_getinteger;
-	s->__sq_getfloat = __sq_getfloat;
-	s->__sq_getbool = __sq_getbool;
-	s->__sq_get = __sq_get;
-	s->__sq_getasset = __sq_getasset;
-	s->__sq_getuserdata = __sq_getuserdata;
-	s->__sq_getvector = __sq_getvector;
-	s->__sq_getthisentity = __sq_getthisentity;
-	s->__sq_getobject = __sq_getobject;
-
-	s->__sq_stackinfos = __sq_stackinfos;
-
-	s->__sq_createuserdata = __sq_createuserdata;
-	s->__sq_setuserdatatypeid = __sq_setuserdatatypeid;
-	s->__sq_getfunction = __sq_getfunction;
-
-	s->__sq_schedule_call_external = AsyncCall_External;
-
-	s->__sq_getentityfrominstance = __sq_getentityfrominstance;
-	s->__sq_GetEntityConstant_CBaseEntity = __sq_GetEntityConstant_CBaseEntity;
-
-	s->__sq_pushnewstructinstance = __sq_pushnewstructinstance;
-	s->__sq_sealstructslot = __sq_sealstructslot;
-}
-
 // Allows for generating squirrelmessages from plugins.
 void AsyncCall_External(ScriptContext context, const char* func_name, SquirrelMessage_External_Pop function, void* userdata)
```

Comment:
this seams to unused?
I like this function so consider keeping it somehow or else I would have to implement it in rrplug :<

Reply:
Honestly I think it's best if the MessageBuffer gets rewritten entirely. Different PR though.
It's pretty easy to implement in plugins though.
```rust
fn RunFrame() {
  for callback in rfCallbacks {
    callback();
  }
}

fn RunNextFrame(callback: fn()) {
  rfCallbacks.push(callback);
}

// ...
RunNextFrame(|| {
  SQObject fn;
  SQRESULT r = sq_getfunction(sqvm, name, &fn, 0);
  if(r == SQRESULT_ERROR) // it's an sqresult iirc
    return; 
  sq_pushobject(sqvm, fn);
  sq_pushroottable(sqvm);
  sq_call(sqvm, 1, false, true);
});
```

Reply:
well it would require each plugin that needs that to add a atomic operation to runframe and keep track of the sqvm pointer which might not be a lot anyway `¯\_(ツ)_/¯`

Reply:
it doesn't have to use an mutex or something like that, the performance overhead is negligible

Reply:
it does or else it won't be thread safe. I would use channel thought.